server/pulp/server/async/tasks.py

```diff
@@ -255,7 +255,7 @@ def apply_async_with_reservation(self, resource_type, resource_id, *args, **kwar
         # To avoid the race condition where __call__ method below is called before
         # this change is propagated to all db nodes, using an 'upsert' here and setting
         # the task state to 'waiting' only on an insert.
-        task_status.save(fields_to_set_on_insert=['state', 'start_time'])
+        task_status.save(fields_to_set_on_insert=['state', 'start_time'], safe=True)
```

Comment:
as @mhrivnak alluded to, safe=True will be completely ignored in newer versions of mongoengine. :disappointed: I see safe=True as an option on the 0.7.10 docs I've built locally, but in the newer docs I think you'll have to use write_concern. See the [docs on save()](http://docs.mongoengine.org/en/latest/apireference.html#mongoengine.Document.save)